currency-rates/internal/queue/producer.go

```diff
@@ -0,0 +1,22 @@
+package queue
+
+import (
+	"github.com/nats-io/nats.go"
+
+	"github.com/RinaDish/currency-rates/tools"
+)
+
+type Queue struct {
+	QueueConn *nats.Conn
+	subscriptionTopicName string
+	logger tools.Logger
+}
+
+
+func NewQueue(nats *nats.Conn, subscriptionTopicName string, logger tools.Logger) (*Queue) {
```

Comment:
Seems like something specific to nats. You can't reuse this for another message queues. Maybe it should accept interface arguments like 
```go
type Broker interface {
     Publish(subj string, data []byte) error
}
```